packages/scripts/src/utils/utils.ts

```diff
@@ -1,7 +1,50 @@
 const hre = require("hardhat");
+import nodeFetch from "node-fetch";
 const { ethers } = hre;
 
 export const increaseEvmTime = async (time: number): Promise<void> => {
   await ethers.provider.send("evm_increaseTime", [time]);
   await ethers.provider.send("evm_mine", []);
 };
+
+export const takeSnapshot = async (): Promise<string> => {
+  const snapshot = await ethers.provider.send("evm_snapshot", []);
+  return snapshot;
+};
+
+export const revertToSnapshot = async (snapshotId: string): Promise<void> => {
+  await ethers.provider.send("evm_revert", [snapshotId]);
+};
+
+export const forkNetwork = (jsonRpcUrl: string, blockNumber?: string): Promise<void> => {
+  return hre.network.provider.request({
+    method: "hardhat_reset",
+    params: [
+      {
+        forking: {
+          jsonRpcUrl,
+          blockNumber,
+        },
+      },
+    ],
+  });
+};
+
+export const getForkChainId = async (jsonRpcUrl: string): Promise<number> => {
```

Comment:
When forking a network hardhat uses the test chain id `31337` instead of fetching the id from the node. This is the only I have found to get the real chain id.

Reply:
maybe its easier to use ethers provider, e.g. something like:

```
import { JsonRpcProvider } from "@ethersproject/providers";
```

and then in `getForkChainId`:

```
    const provider = new JsonRpcProvider(jsonRpcUrl);
    return (await provider.getNetwork()).chainId;
```

Reply:
Nice. That worked, thanks @Reinis-FRP